src/lib/message/L1ToL2Message.ts

```diff
@@ -432,7 +429,20 @@ export class L1ToL2MessageReader extends L1ToL2Message {
     }
 
     // not redeemed, has it now expired
-    if (await this.isExpired()) {
+    try {
+      if (await this.isExpired()) {
```

Comment:
I think we should make `isExpired` atomic with `getSuccessfulRedeem` so that we always have consistent results for `getSuccessfulRedeem`.

We could adjust `isExpired` so that it has a try/catch internally, and rename it to `retryableExists`.

Then adjust `getSuccessfulRedeem` so that we call `retryableExists` somewhere at the start of it. If `retryableExists` is true then we can just return null straight away - since it hasnt been redeemed or expired. If `retryableExists` is false then we either successfully redeemed or we expired. We then look for a successful redeem, if we find it we return the tx receipt, if we dont we return some value that indicates an expiration.

Then we won't check `isExpired/retryableExists` in receiptToStatus, instead we just pass in the receipt, or the value that indicates expiration.